packages/docusaurus/src/commands/start.ts

```diff
@@ -135,15 +137,20 @@ export async function start(
       rewrites: [{from: /\/*/, to: baseUrl}],
     },
     disableHostCheck: true,
-    // Enable overlay on browser. E.g: display errors
-    overlay: true,
+    // Disable overlay on browser since we use CRA's overlay error reporting
+    overlay: false,
     host,
-    before: app => {
+    before: (app, server) => {
       app.use(baseUrl, express.static(path.resolve(siteDir, STATIC_DIR_NAME)));
+
+      // This lets us fetch source contents from webpack for the error overlay
+      app.use(evalSourceMapMiddleware(server));
+      // This lets us open files from the runtime error overlay.
+      app.use(errorOverlayMiddleware());
+
       // TODO: add plugins beforeDevServer and afterDevServer hook
     },
   };
-  WebpackDevServer.addDevServerEntrypoints(config, devServerConfig);
```

Comment:
No longer needed cause we dont want to use the default client from webpack-dev-server. We use CRA's